Sources/PubNub/EventEngine/Subscribe/Effects/SubscribeEffectFactory.swift

```diff
@@ -14,13 +14,13 @@ class SubscribeEffectFactory: EffectHandlerFactory {
   private let session: SessionReplaceable
   private let sessionResponseQueue: DispatchQueue
   private let messageCache: MessageCache
-  private let presenceStateContainer: PresenceStateContainer
+  private let presenceStateContainer: PubNubPresenceStateContainer
   
   init(
     session: SessionReplaceable,
-    sessionResponseQueue: DispatchQueue = .global(qos: .default),
+    sessionResponseQueue: DispatchQueue = .main,
```

Comment:
Is this intentional change or happened during tests?

Reply:
Yeah, it is intentional change. I noticed that `master` implementation provides Subscribe and Heartbeat response on a main thread and this is what I restored to stay compatible. There was a problem with `testUnsubscribeResubscribe()` integration tests that randomly failed. And right now, it's no longer present.